pkg/operator/controller/ingress/controller.go

```diff
@@ -412,19 +412,63 @@ func setDefaultPublishingStrategy(ic *operatorv1.IngressController, infraConfig
 				changed = true
 			}
 
```

Comment:
Is this change fixing a bug? Can you add a comment here, to replace the one you removed?

Reply:
The intention wasn't to fix bugs, but it probably does prevent some weird behavior and could prevent future bugs.  I've added some comments and rewritten the commit message to elaborate on this point.

Reply:
Does [commit "rework GCP logic"](b45521cb8bc9d9b6f04cc6e6c7237e6b68bdada7) belong to this PR? Can it go in the upcoming commit?

Reply:
It does.  Are you suggesting I merge the commits?  I thought it would be much simpler to understand the changes if I first refactored to simplify the existing logic and then added the new logic.